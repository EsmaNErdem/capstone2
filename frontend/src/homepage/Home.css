/* styles.css */
@import url('https://fonts.googleapis.com/css2?family=Alegreya+Sans+SC:wght@500&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Rubik+Doodle+Shadow&display=swap');

.Home {
    padding: 2% 25%;
    display: flex;
    flex-direction: row;
    align-items: center;
    text-align: center;
}

@media (max-width: 1400px) {
  .Home {
    flex-direction: column-reverse;
    padding: 5%;
  }
}

.Home-Card {
    position: relative;
    width: 80vw;
    max-width: 400px;
    height: 600px;
    background-color: floralwhite;
    border-top-right-radius: 10px;
    border-bottom-right-radius: 10px;
    cursor: pointer;
    box-shadow: 10px 0 20px rgb(0, 0, 0, 0.4), inset 10px 0 20px rgb(0, 0, 0, 0.4);
    transform: perspective(2000px) rotateZ(-8deg);
    transform-style: preserve-3d;
    transition: all 1s ease;
    margin: 20px;
}

.Home-Cover .book-headline {
    font-family: 'Rubik Doodle Shadow';
    text-shadow: rgb(140, 116, 19) 1px 0 10px;
    color: darkgoldenrod;
    font-size: 2.5rem;
    font-weight: bold;
    margin-bottom: 2rem;
    position: absolute;
    top: 20%;
    left: 20%;
    z-index: 1;
    backface-visibility: hidden;
}

.Home-Cover {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: #000;
    border-top-right-radius: 10px;
    border-bottom-right-radius: 10px;
    transform-origin: left;
    transform-style: preserve-3d;
    transition: all 1s ease;
}

.Home-Cover img {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    object-fit: cover;
    backface-visibility: hidden;
    border-radius: 10px;
}

.Home-Cover img:nth-child(2) {
    border-top-right-radius: 10px;
    border-bottom-right-radius: 10px;
}

.Home-Cover img:nth-child(2) {
    transform: rotateY(180deg);
}

.Home-Card:hover .Home-Cover {
    transform: rotateY(-135deg);
    box-shadow: 10px 0 20px rgb(0, 0, 0, 0.4);
}

.Home-Card:hover {
    transform: perspective(2000px) rotateZ(-5deg);
    box-shadow: 10px 0 20px rgb(0, 0, 0, 0.4), inset 10px 0 20px rgb(0, 0, 0, 0.4);
}

.Home-Content {
    width: 100%;
    height: 100%;
    padding: 1rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.Home-Content .headline {
    color: #a074f3;
    font-size: 2rem;
    font-weight: bold;
    margin-bottom: 2rem;
    font-family: 'Alegreya Sans SC', sans-serif;
}

.Home-Content .description {
    color: #093b0f;
    font-weight: bold;
    font-size: 1.2rem;
    margin-bottom: 20px;
    font-family: 'Alegreya Sans SC', sans-serif;
}

.homeContainer {
    margin-top: 2rem;
}

.homeContainer .welcomeMessage {
    color: yellowgreen;
    text-shadow: green 1px 0 3px;
    font-family: 'Alegreya Sans SC', sans-serif;
    font-size: 2rem;
    font-weight: bold;
    margin-top: 20px;
}

.homeContainer .browse-button {
    font-family: 'Alegreya Sans SC', sans-serif;
    margin-top: 2rem;
    font-size: 1.5rem;
    color: #a074f3;
    text-shadow: purple 1px 0 3px;
}

.homeContainer .browse-button:hover {
    text-decoration: underline;
}

.authButtons {
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 20px;
    margin-top: 20px;
}

.authButtons .buttons {
    font-family: 'Alegreya Sans SC', sans-serif;
    margin-top: 2rem;
    font-size: 1.5rem;
    color: #a074f3;
    text-shadow: purple 1px 0 3px;
}
.authButtons .buttons:hover {
    text-decoration: underline;
}