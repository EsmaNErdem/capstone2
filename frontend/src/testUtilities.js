import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  firstNsme: "testfirst",
  lastName: "testlast",
  email: "test@test.net",
  img: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, hasLikedReview = () => false, hasLikedBook = () => false, isUserReview = () => false, hasFollowing = () => false, setCurrentUser = u => u }) => (
    <UserContext.Provider value={{ currentUser, hasLikedReview, hasLikedBook, hasFollowing, setCurrentUser, isUserReview }}>
      {children}
    </UserContext.Provider>
);

const CurrUserProvider =
    ({ children, currentUser = demoUser, hasLikedReview = () => false, hasLikedBook = () => false, isUserReview = () => true, hasFollowing = () => false, setCurrentUser = u => u }) => (
    <UserContext.Provider value={{ currentUser, hasLikedReview, hasLikedBook, hasFollowing, setCurrentUser, isUserReview }}>
      {children}
    </UserContext.Provider>
);

const NonUserProvider =
    ({ children, currentUser = null, hasLikedReview = () => false, hasLikedBook = () => false, hasFollowing = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasLikedReview, hasLikedBook, hasFollowing,}}>
      {children}
    </UserContext.Provider>
);

export { UserProvider, CurrUserProvider, NonUserProvider };
