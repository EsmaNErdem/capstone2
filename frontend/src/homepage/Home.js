import React, { useContext } from "react";
import { Link } from "react-router-dom";
import UserContext from "../auth/UserContext";
import { Typography, Button, Container } from '@mui/material';
import "./Home.css"

/**
 * Homepage Component
 * Displays a welcome message with login and signup buttons
 * Routed at /
 * 
 * Routes ==> Home
 */
const Home = ({ login }) => {
  const { currentUser } = useContext(UserContext);

  const LoginTestUser = async () => {
    await login({ username: 'TestUser', password: '123456789'})
  }

  return (
    <div className="Home">
      <div className="Home-Card">
        <div className="Home-Cover">
          <img src="https://media.istockphoto.com/id/453166187/photo/dirty-old-green-book-cover.jpg?s=612x612&w=0&k=20&c=CwPKZXsGQNHUhoH_-tL7yFZT87rW4iDkCJsoIGCZPMI=" alt="home-cover" />
          <img src="https://img.freepik.com/premium-photo/dark-purple-old-velvet-fabric-texture-used-as-background-empty-purple-fabric-background-soft-smooth-textile-material-there-is-space-textx9_661047-2260.jpg"  alt="home=cover-backside"/>
          
          <Typography variant="h3" className="book-headline">
            BOOK CHAT
          </Typography>
        </div>
        <div className="Home-Content">
          <Typography variant="h3" className="headline">
            Book CHAT
          </Typography>
          <Typography variant="body1" className="description">
            Welcome to Book Chat, where book lovers unite to share their passion for reading.
          </Typography>
          <Typography variant="body1" className="description">
            Book Talk is not just an app; it's your gateway to a vibrant community of book lovers. Immerse yourself in the world of literature, connect with like-minded readers, and elevate your reading experience.
            Write your opions and take aways on your favorites books and read others books reviews to open your understanding
          </Typography>
          <Typography variant="body1" className="description">
            Ground Rules: Be Kind
          </Typography>
        </div>
      </div>
      <Container className="homeContainer">
        {currentUser ? (
          <>
            <Typography variant="h4" className="welcomeMessage">
              Welcome, {currentUser.username}!
            </Typography>
            <Button component={Link} to="/books" href="#text-buttons" className="browse-button">
              Browse Books
            </Button>
          </>
        ) : (
          <div className="authButtons">
            <Typography variant="h4" className="welcomeMessage">
              Sign Up to look around book and write reviews
            </Typography>
           <div>
            <Button component={Link} to="/login"  href="#text-buttons"
            className="buttons">
                Login
            </Button>
            <Button component={Link} to="/signup" href="#text-buttons" className="buttons">
                Sign Up
              </Button>
            <Button onClick={LoginTestUser} className="buttons">
                Login As a TestUser
              </Button>
           </div>
          </div>
        )}
      </Container>
    </div>
  );
};

export default Home;