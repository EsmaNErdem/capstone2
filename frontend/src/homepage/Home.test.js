import { render } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom';
import { UserProvider, NonUserProvider } from "../testUtilities";
import Home from "./Home"


test("renders without crashing", () => {
    render(
        <MemoryRouter>
            <NonUserProvider>
                <Home />
            </NonUserProvider>
        </MemoryRouter>
    );
});

test("it renders and matches with snaphot", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <NonUserProvider>
                <Home />
            </NonUserProvider>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

test("it renders and matches with snaphot", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <NonUserProvider>
                <Home />
            </NonUserProvider>
        </MemoryRouter>
        
    );
    expect(asFragment()).toMatchSnapshot();
});

test("it renders welcome message for non-user", () => {
    const { getByText } = render(
        <MemoryRouter>
            <NonUserProvider>
                <Home />
            </NonUserProvider>
        </MemoryRouter>
    );
    expect(getByText("Welcome to Book Chat, where book lovers unite to share their passion for reading.")).toBeInTheDocument();
    expect(getByText("Login")).toBeInTheDocument();
});

test("it renders welcome message for user", () => {
    const { getByText } = render(
        <MemoryRouter>
            <UserProvider>
                <Home />
            </UserProvider>
        </MemoryRouter>
    );
    expect(getByText("Welcome to Book Chat, where book lovers unite to share their passion for reading.")).toBeInTheDocument();
    expect(getByText("Welcome, testuser!")).toBeInTheDocument();
});
