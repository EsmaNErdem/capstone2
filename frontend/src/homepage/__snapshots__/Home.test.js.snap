// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches with snaphot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <div
      class="Home-Card"
    >
      <div
        class="Home-Cover"
      >
        <img
          alt="home-cover"
          src="https://media.istockphoto.com/id/453166187/photo/dirty-old-green-book-cover.jpg?s=612x612&w=0&k=20&c=CwPKZXsGQNHUhoH_-tL7yFZT87rW4iDkCJsoIGCZPMI="
        />
        <img
          alt="home=cover-backside"
          src="https://img.freepik.com/premium-photo/dark-purple-old-velvet-fabric-texture-used-as-background-empty-purple-fabric-background-soft-smooth-textile-material-there-is-space-textx9_661047-2260.jpg"
        />
        <h3
          class="MuiTypography-root MuiTypography-h3 book-headline css-gepadz-MuiTypography-root"
        >
          BOOK CHAT
        </h3>
      </div>
      <div
        class="Home-Content"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 headline css-gepadz-MuiTypography-root"
        >
          Book CHAT
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body1 description css-ahj2mt-MuiTypography-root"
        >
          Welcome to Book Chat, where book lovers unite to share their passion for reading.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 description css-ahj2mt-MuiTypography-root"
        >
          Book Talk is not just an app; it's your gateway to a vibrant community of book lovers. Immerse yourself in the world of literature, connect with like-minded readers, and elevate your reading experience. Write your opions and take aways on your favorites books and read others books reviews to open your understanding
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 description css-ahj2mt-MuiTypography-root"
        >
          Ground Rules: Be Kind
        </p>
      </div>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg homeContainer css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="authButtons"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 welcomeMessage css-5lbw0b-MuiTypography-root"
        >
          Sign Up to look around book and write reviews
        </h4>
        <div>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium buttons css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            href="/login"
            tabindex="0"
          >
            Login
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium buttons css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            href="/signup"
            tabindex="0"
          >
            Sign Up
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium buttons css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Login As a TestUser
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it renders and matches with snaphot 2`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <div
      class="Home-Card"
    >
      <div
        class="Home-Cover"
      >
        <img
          alt="home-cover"
          src="https://media.istockphoto.com/id/453166187/photo/dirty-old-green-book-cover.jpg?s=612x612&w=0&k=20&c=CwPKZXsGQNHUhoH_-tL7yFZT87rW4iDkCJsoIGCZPMI="
        />
        <img
          alt="home=cover-backside"
          src="https://img.freepik.com/premium-photo/dark-purple-old-velvet-fabric-texture-used-as-background-empty-purple-fabric-background-soft-smooth-textile-material-there-is-space-textx9_661047-2260.jpg"
        />
        <h3
          class="MuiTypography-root MuiTypography-h3 book-headline css-gepadz-MuiTypography-root"
        >
          BOOK CHAT
        </h3>
      </div>
      <div
        class="Home-Content"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 headline css-gepadz-MuiTypography-root"
        >
          Book CHAT
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body1 description css-ahj2mt-MuiTypography-root"
        >
          Welcome to Book Chat, where book lovers unite to share their passion for reading.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 description css-ahj2mt-MuiTypography-root"
        >
          Book Talk is not just an app; it's your gateway to a vibrant community of book lovers. Immerse yourself in the world of literature, connect with like-minded readers, and elevate your reading experience. Write your opions and take aways on your favorites books and read others books reviews to open your understanding
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 description css-ahj2mt-MuiTypography-root"
        >
          Ground Rules: Be Kind
        </p>
      </div>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg homeContainer css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="authButtons"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 welcomeMessage css-5lbw0b-MuiTypography-root"
        >
          Sign Up to look around book and write reviews
        </h4>
        <div>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium buttons css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            href="/login"
            tabindex="0"
          >
            Login
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium buttons css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            href="/signup"
            tabindex="0"
          >
            Sign Up
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium buttons css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Login As a TestUser
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
